lib/co_aspects/attacher.rb

```diff
@@ -0,0 +1,42 @@
+require 'active_support/core_ext/string/inflections'
+
+module CoAspects
+  class Attacher
+    def initialize
+      @enabled = true
+      @pending = []
+    end
+
+    def add(method_name, args, block)
+      aspect_class = aspect_from_method(method_name)
+      @pending << Annotation.new(aspect_class, args, block)
+    end
+
+    def attach(klass, method_name)
+      if @enabled
+        @enabled = false
+        @pending.each do |pending|
+          pending.aspect.apply klass,
+            method: method_name,
+            args: pending.args,
+            block: pending.block
+        end
+        @enabled = true
+        @pending = []
+      end
+    end
```

Comment:
What do you think about adding a private method to handle the blocking? Something like:
```ruby
def attach(klass, method_name)
  blocking do
    @pending.each do |pending|
      pending.aspect.apply klass,
        method: method_name,
        args: pending.args,
        block: pending.block
    end
    @pending = []
  end
end

private

def blocking
  @enabled = true unless defined?(@enabled)
  if @enabled
    @enabled = false
    yield
    @enabled = true
  end
end
```
This way you can leave the actual logic a bit more clear (and you could remove the `@enabled = true` from the initializer)